HPE3PARPSToolkit-CLI.psm1

```diff
@@ -34380,11 +34379,2491 @@ Function Stop-3parWsapi()
+ }
+
+ $Result = Invoke-3parCLICmd -Connection $SANConnection -cmds  $Cmd
+ Write-DebugLog "Executing Function : Set-3parDomain Command" INFO: 
+ IF ([System.String]::IsNullOrEmpty($Domain))
+ {
+	$Result = "Working domain is set to no current domain."
+	Return $Result
+ }
+ else
+ {
+	Return $Result
+ }
+ 
+} ##  End-of Set-3parDomain
+
+##########################################################################
+######################### FUNCTION Get-3parDomain #########################
+##########################################################################
+Function Get-3parDomain()
+{
+<#
+  .SYNOPSIS
+   showdomain - Show information about domains in the system.
+
+  .DESCRIPTION
+   The showdomain command displays a list of domains in a system.
+
+  .EXAMPLE
+
+  .PARAMETER D
+   Specifies that detailed information is displayed.
+
+  .Notes
+    NAME: Get-3parDomain
+    LASTEDIT 19-03-2019 13:34:53
+    KEYWORDS: 3parVersion
+  
+  .Link
+    Http://www.hpe.com
+
+ #Requires PS -Version 3.0
+#>
+[CmdletBinding()]
+ param(
+	[Parameter(Position=0, Mandatory=$false, ValueFromPipeline=$true)]
+	[switch]
+	$D,
+
+	[Parameter(Position=1, Mandatory=$false, ValueFromPipeline=$true)]
+	$SANConnection = $global:SANConnection
+ )
+
+ Write-DebugLog "Start: In Get-3parDomain - validating input values" $Debug 
+ #check if connection object contents are null/empty
+ if(!$SANConnection)
+ {
+	#check if connection object contents are null/empty
+	$Validate1 = Test-ConnectionObject $SANConnection
+	if($Validate1 -eq "Failed")
+	{
+		#check if global connection object contents are null/empty
+		$Validate2 = Test-ConnectionObject $global:SANConnection
+		if($Validate2 -eq "Failed")
+		{
+			Write-DebugLog "Connection object is null/empty or Connection object UserName,password,IPAaddress are null/empty. Create a valid connection object using New-SANConnection" " ERR: "
+			Write-DebugLog "Stop: Exiting Get-3parDomain since SAN connection object values are null/empty" $Debug 
+			Return "FAILURE : Exiting Get-3parDomain since SAN connection object values are null/empty"
+		}
+	}
+ }
+
+ $plinkresult = Test-PARCli -SANConnection $SANConnection
+ if($plinkresult -match "FAILURE :")
+ {
+   write-debuglog "$plinkresult"
+   Return $plinkresult
+ }
+
+ $Cmd = " showdomain "
+
+ if($D)
+ {
+	$Cmd += " -d "
+ }
+
+ $Result = Invoke-3parCLICmd -Connection $SANConnection -cmds  $Cmd
+ Write-DebugLog "Executing Function : Get-3parDomain Command -->" INFO: 
+  
+ if($Result.count -gt 1)
+ {
+	$Cnt = $Result.count
+		
+ 	$tempFile = [IO.Path]::GetTempFileName()
+	$LastItem = $Result.Count -2  
+	
+	foreach ($s in  $Result[0..$LastItem] )
+	{		
+		$s= [regex]::Replace($s,"^ ","")			
+		$s= [regex]::Replace($s," +",",")	
+		$s= [regex]::Replace($s,"-","")
+		$s= $s.Trim() 
+		$temp1 = $s -replace 'CreationTime','Date,Time,Zone'
+		$s = $temp1		
+		Add-Content -Path $tempfile -Value $s				
+	}
+	Import-Csv $tempFile 
+	del $tempFile 	
+ }
+ 
+ if($Result.count -gt 1)
```

Comment:
The same condition check "if($Result.count -gt 1)" is executed above (line #34555) . Can you give an else clause to the above if($Result.count -gt 1) check instead of the repetition.

Reply:
Can you include " if($Result.count -gt 1) " if required. Otherwise simply return the $Result as you have done for below PS cmdlets.

Reply:
Added.